Libraries/LibWeb/CSS/Interpolation.cpp

```diff
@@ -461,6 +461,141 @@ static RefPtr<StyleValue const> interpolate_rotate(DOM::Element& element, Calcul
         { interpolated_x_axis, interpolated_y_axis, interpolated_z_axis, interpolated_angle });
 }
 
+static Optional<GridTrackSizeList> interpolate_grid_track_size_list(CalculationContext const& calculation_context, GridTrackSizeList const& from, GridTrackSizeList const& to, float delta)
+{
+    auto interpolate_css_size = [&](Size const& from_size, Size const& to_size) -> Size {
+        if (from_size.is_length_percentage() && to_size.is_length_percentage()) {
+            auto interpolated_length = interpolate_length_percentage(calculation_context, from_size.length_percentage(), to_size.length_percentage(), delta);
+            return Size::make_length_percentage(*interpolated_length);
+        }
+
+        if (from_size.type() != to_size.type())
+            return delta < 0.5f ? from_size : to_size;
+
+        switch (from_size.type()) {
+        case Size::Type::FitContent: {
+            if (!from_size.fit_content_available_space().has_value() || !to_size.fit_content_available_space().has_value())
+                break;
+            auto interpolated_available_space = interpolate_length_percentage(calculation_context, *from_size.fit_content_available_space(), *to_size.fit_content_available_space(), delta);
+            if (!interpolated_available_space.has_value())
+                break;
+            return Size::make_fit_content(interpolated_available_space.release_value());
+        }
+        default:
+            break;
+        }
+
+        return delta < 0.5f ? from_size : to_size;
+    };
+
+    auto interpolate_grid_size = [&](GridSize const& from_grid_size, GridSize const& to_grid_size) -> GridSize {
+        if (from_grid_size.is_flexible_length() || to_grid_size.is_flexible_length()) {
+            if (!from_grid_size.is_flexible_length() && from_grid_size.css_size().is_auto() && !to_grid_size.is_flexible_length() && to_grid_size.css_size().is_auto())
```

Comment:
This condition seems a bit wrong - either `from_grid_size` or `to_grid_size` must have `is_flexible_length() == true` because of the line above.

Reply:
This condition is clearly never true :sweat_smile: I think this is just the result of me rushing through a refactor and getting it a bit wrong.

My latest push removes this check.